core/app/models/spree/stock/simple_coordinator.rb

```diff
@@ -52,7 +52,9 @@ def build_shipments
         on_hand_packages, backordered_packages, leftover = @allocator.allocate_inventory(@desired)
 
         unless leftover.empty?
-          raise Spree::Order::InsufficientStock
+          item_names = leftover.variants.map(&:name).to_sentence
+          message = I18n.t('spree.inventory_error_flash_for_insufficient_quantity', names: item_names)
+          raise Spree::Order::InsufficientStock.new(message)
```

Comment:
What about enhancing this spec https://github.com/solidusio/solidus/blob/0f76bac523d557d3aff430c50f55d8fa832be5c5/core/spec/models/spree/stock/simple_coordinator_spec.rb#L130 to check also for the message?